lib/controllers/git-tab-controller.js

```diff
@@ -269,26 +267,8 @@ export default class GitTabController {
   }
 
   @autobind
-  async commit(message) {
-    try {
-      await this.getActiveRepository().commit(message, {amend: this.props.isAmending});
-      this.setAmending(false);
-      this.props.destroyFilePatchPaneItems({onlyStaged: true});
-    } catch (e) {
-      if (e.code === 'ECONFLICT') {
-        this.props.notificationManager.addError(
-          'Cannot commit without resolving all the merge conflicts first.',
-          {dismissable: true},
-        );
-      } else {
-        throw e;
-      }
-    }
-  }
-
-  @autobind
-  setAmending(isAmending) {
-    this.props.didChangeAmending(isAmending);
+  commit(message) {
+    return this.getActiveRepository().commit(message, {amend: this.isAmending()});
```

Comment:
Since `this.isAmending()` calls out to `this.getActiveRepository()` itself, could we move the boolean into the `Repository#commit` method instead of passing it in as an option?